src/components/Input.jsx

```diff
@@ -64,7 +64,7 @@ const Input = forwardRef(
 
     const handleOnBlur = e => {
       if (!disableTrimOnBlur) {
-        e.target.value = value.trim();
+        e.target.value = value?.trim();
```

Comment:
The issue is not related to `value` being `undefined`, it's when an input type is `number`. 

This change would still fail since the `number?.trim()` would still throw `trim is not a function` error.

I think a solution would be to not trim when the `value` is not a string.